imagersite/imagersite/views.py

```diff
@@ -0,0 +1,37 @@
+from __future__ import unicode_literals
+from django.shortcuts import render, render_to_response
+from django.http import HttpResponse
+from django.template import Template
+from django.template import loader
+from django.views.generic import TemplateView
+from imager_images.models import Photo, Album
+from django.contrib.auth.models import User
+from django.conf import settings
+
+
+def home_view(request):
+    try:
+        pic_url = Photo.objects.filter(
+                  privacy='PU').order_by('?').first().file
+    except AttributeError:
+        pic_url = 'demo.jpg'
+    return render(request, 'home.html', {'pic_url': pic_url})
+
+
+def profile_view(request, username):
+    user = User.objects.filter(username=username).first()
+    album_count = Album.objects.filter(owner=user).count()
+    photo_count = Photo.objects.filter(owner=user).count()
+    return render(request, 'profile.html',
+                 {'album_count': album_count, 'photo_count': photo_count})
+
+
+def photos_view(request, username):
+    user = User.objects.filter(username=username).first()
+    photos = Photo.objects.filter(owner=user).order_by('date_uploaded')
```

Comment:
You can automatically get the photos owned by the user by using the `related_name` of the ForeignKey field on the photo object.  If you haven't set one, then Django gives you the attribute for free as `<model_name>_set`:

```python
photos = user.photo_set.all().order_by('date_uploaded')
```